contrib/scripts/aws-iam-create-yubikey-mfa.sh

```diff
@@ -1,28 +1,23 @@
 #!/bin/sh
 # Adds a Yubikey TOTP device to IAM
 
-set -eu
+set -e
 
 if [ -n "${AWS_SESSION_TOKEN:-}" ]; then
   echo "aws-vault must be run without a STS session, please run it with the --no-session flag" >&2
   exit 1
 fi
 
-cleanup()
-{
-  if [ -z "$OUTFILE" ]; then
-    rm "$OUTFILE"
-  fi
-}
+cleanup() { rm -f "$OUTFILE"; }
 trap cleanup EXIT
```

Comment:
you could do the trap _after_ setting `OUTFILE`

Reply:
I was thinking about that, but in that case, temp file would exist longer time leaving more time for a potential attacker to find it and later read it. That's the same reason `mktemp` and also `rm` is straight before and after the command generating TOTP secret. `rm` would otherwise done on `trap`. what do you think?

Reply:
You can then use `declare`.

Reply:
`declare` did not seem to help. I have put `-u` `(nounset)` back along with default for `OUTFILE`, which seems to work ok. `-o pipefail` is a bashism and doesn't work in `dash`, so I would rather not put that in.